core/trino-main/src/test/java/io/trino/operator/scalar/TestMathFunctions.java

```diff
@@ -3573,4 +3571,16 @@ public void testWilsonInterval()
         assertThat(assertions.function("wilson_interval_upper", "1250", "1310", "1.96e0"))
                 .isEqualTo(0.9642524717143908);
     }
+
+    @Test
+    public void testBlackScholesEuroVanillaCall() {
```

Comment:
Can we try with additional edge case values - maybe if one of them could be set as 0.

Reply:
I have added more tests which should cover the edge cases.